source/common/tracing/lightstep_tracer_impl.cc

```diff
@@ -14,14 +14,33 @@
 
 namespace Tracing {
 
-LightStepSpan::LightStepSpan(lightstep::Span& span) : span_(span) {}
+LightStepSpan::LightStepSpan(lightstep::Span& span, lightstep::Tracer& tracer)
+    : span_(span), tracer_(tracer) {}
 
 void LightStepSpan::finishSpan() { span_.Finish(); }
 
 void LightStepSpan::setTag(const std::string& name, const std::string& value) {
   span_.SetTag(name, value);
 }
 
+void LightStepSpan::injectContext(Http::HeaderMap& request_headers) {
+  lightstep::BinaryCarrier ctx;
+  tracer_.Inject(context(), lightstep::CarrierFormat::LightStepBinaryCarrier,
+                 lightstep::ProtoWriter(&ctx));
+  const std::string current_span_context = ctx.SerializeAsString();
+  request_headers.insertOtSpanContext().value(
+      Base64::encode(current_span_context.c_str(), current_span_context.length()));
+}
+
+SpanPtr LightStepSpan::spawnChild(const std::string& name, SystemTime start_time) {
+  SpanPtr child_span;
+  lightstep::Span ls_span = tracer_.StartSpan(
+      name, {lightstep::ChildOf(span_.context()), lightstep::StartTimestamp(start_time)});
+  child_span.reset(new LightStepSpan(ls_span, tracer_));
+
+  return std::move(child_span);
```

Comment:
can we? this seems to be a common pattern adopted elsewhere, but I'm not familiar enough with the ramifications.